dapp/pages/stake.js

```diff
@@ -200,7 +202,20 @@ const Stake = ({ locale, onLocale, rpcProvider, isMobile }) => {
         }
         tokenToStakeDecimalsCall={ognContract.decimals}
         stakeFunctionCall={async (stakeAmount) => {
-          return ognStaking.stake(stakeAmount, selectedDuration)
+          const stakeAmountString = formatBn(stakeAmount, 18)
```

Comment:
I think we should be able to get rid of the new Web3 dependency and use ethers.js. See https://docs.ethers.io/v5/single-page/#/v5/api/utils/abi/interface/
I haven't tested but something like this should work:
```
const interface = ognStaking.interface
const fragment = ognStaking.interface.getFunction("stakeWithSender(uint256, uint256)")
const fnSig = interface.getSighash(fragment)
const params = interface.encodeFunctionData("stakeWithSender(uint256, uint256)", [stakeAmountString, selectedDuration])
```

Reply:
great thanks using that